internal/users/db/testutils.go

```diff
@@ -0,0 +1,161 @@
+package db
+
+// All those functions and methods are only for tests.
+// They are not exported, and guarded by testing assertions.
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/testsdetection"
+	"github.com/ubuntu/authd/log"
+	"gopkg.in/yaml.v3"
+)
+
+// Z_ForTests_DumpNormalizedYAML gets the content of the database, normalizes it
+// (so that it can be compared with a golden file) and returns it as a YAML string.
+//
+// nolint:revive,nolintlint // We want to use underscores in the function name here.
+func Z_ForTests_DumpNormalizedYAML(t *testing.T, c *Manager) string {
```

Comment:
Just to be safe: I remember you mentioning that these symbols are left out of the release code, but what happens to the dependencies? Are they still included in the package or they are also left out? Here you are importing the `testing` package inside the `db` one and that's something we avoid at all costs... If it's not included in the release, I guess we are fine, but if it is, we should probably go back to return errors and let the test function deal with them.

Reply:
It's not in the binary:

```
go build -o authd ./cmd/authd && go tool nm authd | grep Z_ForTests
```

Reply:
I'm aware that the function isn't part of it, but does it pull the `testing` package into the binary even though it is trimmed out?

Reply:
Ah, I understand what you mean now. Yes, there are indeed some symbols from the testing package:

```
go build -o authd ./cmd/authd && go tool nm authd | grep testing
 1234e28 d os.testingForceReadDirLstat
 11d6160 d testing..inittask
  d48f60 r testing..stmp_19
  d48f70 r testing..stmp_20
  d472fc r testing..stmp_60
  97a240 t testing.init
 1210ad8 d testing.supportedTypes
```

I'll change the functions to return errors.

Reply:
Actually, that's probably something for a follow-up task, because the `testing` package is already used in that file on `main`: https://github.com/ubuntu/authd/blob/0b9e98f5b70e311c07c575cce73099418b56fbdb/internal/users/cache/testutils.go#L13-L13

Reply:
The only line I see `testing` being used there is here: https://github.com/ubuntu/authd/blob/0b9e98f5b70e311c07c575cce73099418b56fbdb/internal/users/cache/testutils.go#L157

So the problem is the same

Reply:
Correct

Reply:
Ok, so we need to remove it from that function as well, right?